src/library/scala/jdk/javaapi/FutureConverters.scala

```diff
@@ -70,9 +70,13 @@ object FutureConverters {
       case cf: CF[T] => cf.wrapped
       // in theory not safe (could be `class C extends Future[A] with CompletionStage[B]`):
       case f: Future[T @unchecked] => f
+      case cf: CompletableFuture[T @unchecked] if cf.isDone && !cf.isCompletedExceptionally =>
```

Comment:
FYI @He-Pin
JDK 11 has MinimalStage which is a CompletableFuture (also CompletionStage), with UnsupportedOperationException in lot of method bodies (like isDone).
With this change we throw from here at line 73 if encounered a MinimalStage.

I am not claiming this is a bug, or MinimalStage is flawed; neither know whether it is a valid case to have MinimalStage flying around, just letting you know.
Don't shoot the messenger please ;)
Others also implement safety measures: https://github.com/Kotlin/kotlinx.coroutines/issues/2456

Reply:
It seems play framework had to work around it: https://github.com/playframework/playframework/pull/11993